### PR TITLE
chore: change positional arguments to kwargs

### LINKED ISSUE #1562: Ban usage of positional arguments
When working with real software positional arguments are very inconvenient.

* They are not self describing, so it's just harder to read and understand them.
* When messed up by the user the feedback is poor, as it's hard to determine which thing exactly is missing and where.
* (when available) shell autocompletion can't do as good job with them.
* (AFAIU) They don't support environment variables
* They can't be reodered, which is very handy for scripting, or when one wrote a long command invocation, and forgot only one argument somewhere.

The only time when positional arguments are OK is when the command really does have a one central argument that is very primary and mandatory. Like `cat` takes file paths. It would be somewhat annoying the have to do `cat --path somefile --path someotherfile`. But that's usually only a concern for commands really meant for short-form interactive use.

I'd like us to stop using positional arguments.

### PR DESCRIPTION
**Pull request by GetPsyched**
Fixes #1562 

Please let me know if these changes are appropriate so that I can complete them for the other command definitions!

### CODE REVIEW

**Review by Maan2003 — COMMENTED** [review]
LGTM

**Comment by GetPsyched** [comment]
I was wondering what should the abbreviated flag be for any given kwarg. What's the general flow to follow when choosing which arguments get what abbreviated flag while also making sure there are no conflicts?

**Review by dpc — DISMISSED** [review]

**Comment by Maan2003** [comment]
> I was wondering what should the abbreviated flag be for any given kwarg. What's the general flow to follow when choosing which arguments get what abbreviated flag while also making sure there are no conflicts?

lets just skip it for now. The short option is currently only being used once(gateway-cli address).

**Comment by justinmoon** [comment]
Should we use kwargs for commands with 1 argument? This would seem kind of annoying to me ... but it makes sense to commands with multiple arguments. Thought @dpc?

**Comment by dpc** [comment]
I'd say 99.9% named arguments. Give them `clap(env = "...")` so if it annoys anyone, they should just `export -x FM_...=value` and have it even better.

**Comment by GetPsyched** [comment]
In which scenarios should I add the env alternative? IMO, having it set for context specific stuff to avoid repetition would be ideal.

**Comment by GetPsyched** [comment]
I think that would be all the commands covered! Please mention if I've missed any commands/attributes that need to be switched to kwargs, I'll fix it right away.

File: client/cli/src/lib.rs [hunk 1]
```diff
@@ -358,48 +358,53 @@ enum Command {
     /// Send direct method call to the API, waiting for all peers to agree on a
     /// response
     Api {
+        #[clap(long)]
         method: String,
         /// JSON args that will be serialized and send with the request
-        #[clap(default_value = "null")]
+        #[clap(long, default_value = "null")]
         arg: String,
```
Comment (Maan2003) [inline client/cli/src/lib.rs:365]: This should be an exception. Positional arguments here feel natural

File: client/cli/src/lib.rs [hunk 2]
```diff
@@ -409,27 +414,34 @@ enum Command {
...
+    WaitInvoice {
+        #[clap(long)]
+        invoice: lightning_invoice::Invoice,
+    },
 
     /// Wait for the fed to reach a consensus block height
-    WaitBlockHeight { height: u64 },
+    WaitBlockHeight {
+        #[clap(long)]
+        height: u64,
```
Comment (Maan2003) [inline client/cli/src/lib.rs:434]: Wait block height is sufficient, we don't need extra --height

**Comment by dpc** [comment]
> In which scenarios should I add the env alternative? IMO, having it set for context specific stuff to avoid repetition would be ideal.

Always use for passing secrets, otherwise arguments that are often used.

**Review by dpc — APPROVED** [review]

**Review by justinmoon — APPROVED** [review]

**Comment by Maan2003** [comment]
I am surprised that this is not failing CLI tests :eyes:

**Pull request merged**